cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaMetadataStateManagerTest.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import io.strimzi.api.kafka.model.kafka.Kafka;
+import io.strimzi.api.kafka.model.kafka.KafkaBuilder;
+import io.strimzi.api.kafka.model.kafka.KafkaMetadataState;
+import io.strimzi.api.kafka.model.kafka.KafkaStatus;
+import io.strimzi.api.kafka.model.kafka.listener.GenericKafkaListenerBuilder;
+import io.strimzi.api.kafka.model.kafka.listener.KafkaListenerType;
+import io.strimzi.operator.common.Annotations;
+import io.strimzi.operator.common.Reconciliation;
+
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.KRaft;
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.KRaftDualWriting;
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.KRaftPostMigration;
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.PreKRaft;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.KRaftMigration;
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.ZooKeeper;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class KafkaMetadataStateManagerTest {
```

Comment:
add Javadoc

Reply:
Isn't it fixed yet?

Reply:
This add Javadoc comment was I guess for `checkTransition()` method. Do we need javadoc for the class also?

Reply:
I think so, even just one line saying what is going to test.

Reply:
Done